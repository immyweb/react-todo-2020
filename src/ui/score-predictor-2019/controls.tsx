import * as React from 'react';

import Icon from '../icon/icon';
import { colours } from '../../styles/colours';

import { NavHolder, Nav, PageIndex, CurrentNum } from './styles';

interface IProps {
  primary: string;
  currentSlide: number;
  slideCount: number;
  onTogglePrev(): void;
  onToggleNext(): void;
}

const Controls: React.FC<IProps> = ({
  primary,
  currentSlide,
  slideCount,
  onTogglePrev,
  onToggleNext,
}) => {
  const firstPage: boolean = currentSlide === 0 ? true : false;
  const lastPage: boolean = currentSlide === slideCount - 1 ? true : false;
  return (
    <NavHolder>
      <Nav onClick={onTogglePrev} disabled={firstPage}>
        <Icon
          iconName="chevron-solid-left"
          iconColour={firstPage ? colours.$lightGrey : colours.$offBlack}
          iconSize={18}
        />
      </Nav>
      <PageIndex>
        <CurrentNum colour={primary}>{`${currentSlide + 1}`}</CurrentNum> /{' '}
        <span className="control__slidecount">{`${slideCount}`}</span>
      </PageIndex>
      <Nav onClick={onToggleNext} disabled={lastPage}>
        <Icon
          iconName="chevron-solid-right"
          iconColour={lastPage ? colours.$lightGrey : colours.$offBlack}
          iconSize={18}
        />
      </Nav>
    </NavHolder>
  );
};

export default Controls;
