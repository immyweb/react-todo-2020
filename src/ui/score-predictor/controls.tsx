import * as React from 'react';
import { colours } from '../../styles/colours';
import Icon from '../icon/icon';
import { NavHolder, Nav, PageIndex, CurrentNum, SlideCount } from './styles';

interface IProps {
  currentSlide: number;
  slideCount: number;
  onTogglePrev(): void;
  onToggleNext(): void;
}

const Controls: React.SFC<IProps> = ({ currentSlide, slideCount, onTogglePrev, onToggleNext }) => {
  const firstPage: boolean = currentSlide === 0 ? true : false;
  const lastPage: boolean = currentSlide === slideCount - 1 ? true : false;

  return (
    <NavHolder>
      <Nav onClick={onTogglePrev} disabled={firstPage}>
        <Icon
          iconName="chevron-solid-left"
          iconColour={firstPage ? colours.$lightGrey : colours.$offBlack}
          iconSize={18}
        />
      </Nav>
      <PageIndex>
        <CurrentNum>{`${currentSlide + 1}`}</CurrentNum> /{' '}
        <SlideCount>{`${slideCount}`}</SlideCount>
      </PageIndex>
      <Nav onClick={onToggleNext} disabled={lastPage}>
        <Icon
          iconName="chevron-solid-right"
          iconColour={lastPage ? colours.$lightGrey : colours.$offBlack}
          iconSize={18}
        />
      </Nav>
    </NavHolder>
  );
};

export default Controls;
