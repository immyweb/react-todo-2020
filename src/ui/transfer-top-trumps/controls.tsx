import * as React from 'react';

import ButtonCta from './button-cta';

import { colours } from '../../styles/colours';
import { ControlsHolder } from './styles';

import { Vote, IPlayer } from './types';

interface IProps {
  currentPlayer: IPlayer;
  className?: string;
  onCastVote(player: IPlayer, vote: Vote): void;
}

export const Controls: React.SFC<IProps> = ({ currentPlayer, onCastVote }) => {
  return (
    <ControlsHolder>
      <ButtonCta
        className={'ttt-button-no'}
        bkgndTop={colours.$lightRed}
        bkgndBottom={colours.$darkRed}
        iconName={'cross-solid'}
        onClick={() => onCastVote(currentPlayer, Vote.no)}
      />
      <ButtonCta
        className={'ttt-button-yes'}
        bkgndTop={colours.$lightGreen}
        bkgndBottom={colours.$darkGreen}
        iconName={'tick-solid'}
        onClick={() => onCastVote(currentPlayer, Vote.yes)}
      />
    </ControlsHolder>
  );
};
