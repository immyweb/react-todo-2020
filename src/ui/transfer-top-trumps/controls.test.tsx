import * as React from 'react';
import { shallow } from 'enzyme';

import { Controls } from './controls';
import ButtonCta from './button-cta';

const currentPlayer = {
  id: 'l_messi',
  fname: 'lionel',
  sName: 'messi',
  imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg',
  imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg',
  club: 'Barcelona',
  clubBadge: 'https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png',
  age: 31,
  fee: 'Â£100m',
  position: 'CF',
  idx: 0,
};

function shallowRender(props?: any) {
  return shallow(<Controls currentPlayer={currentPlayer} onCastVote={() => {}} {...props} />);
}

describe('<Controls />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onCastVote when No button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onCastVote: fn });
    const btn = wrapper.find(ButtonCta).first();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('calls onCastVote when Yes button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onCastVote: fn });
    const btn = wrapper.find(ButtonCta).last();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
