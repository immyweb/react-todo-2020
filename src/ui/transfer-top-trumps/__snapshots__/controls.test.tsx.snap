// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentPlayer={
      Object {
        "age": 31,
        "club": "Barcelona",
        "clubBadge": "https://www.thesun.co.uk/wp-content/uploads/2018/07/chelsea_@2x.png",
        "fee": "Â£100m",
        "fname": "lionel",
        "id": "l_messi",
        "idx": 0,
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_desktop_@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/07/WILLIAN_mobile_@2x.jpg",
        "position": "CF",
        "sName": "messi",
      }
    }
    onCastVote={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ButtonCta
          bkgndBottom="#9f041b"
          bkgndTop="#ed283a"
          className="ttt-button-no"
          iconName="cross-solid"
          onClick={[Function]}
        />,
        <ButtonCta
          bkgndBottom="#6cb71b"
          bkgndTop="#a5dc39"
          className="ttt-button-yes"
          iconName="tick-solid"
          onClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bkgndBottom": "#9f041b",
          "bkgndTop": "#ed283a",
          "className": "ttt-button-no",
          "iconName": "cross-solid",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bkgndBottom": "#6cb71b",
          "bkgndTop": "#a5dc39",
          "className": "ttt-button-yes",
          "iconName": "tick-solid",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ButtonCta
            bkgndBottom="#9f041b"
            bkgndTop="#ed283a"
            className="ttt-button-no"
            iconName="cross-solid"
            onClick={[Function]}
          />,
          <ButtonCta
            bkgndBottom="#6cb71b"
            bkgndTop="#a5dc39"
            className="ttt-button-yes"
            iconName="tick-solid"
            onClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bkgndBottom": "#9f041b",
            "bkgndTop": "#ed283a",
            "className": "ttt-button-no",
            "iconName": "cross-solid",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bkgndBottom": "#6cb71b",
            "bkgndTop": "#a5dc39",
            "className": "ttt-button-yes",
            "iconName": "tick-solid",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
